### PR TITLE
.ci: don't notify of packaging failure on PRs

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Don't send notifications if packaging fails in a PR. Notifications are only relevant for branch builds; developers should check PR status directly.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

N/A

## Related issues

N/A

### CODE REVIEW

**Review by simitt — APPROVED** [review]
Thanks!

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -118,8 +118,10 @@ pipeline {
           }
           post {
             failure {
-              notifyStatus(subject: "[${env.REPO}@${env.BRANCH_NAME}] package failed.",
-                           body: 'Contact the Productivity team [#observablt-robots] if you need further assistance.')
+              if (!isPR()) {
```
Comment (v1v) [inline .ci/packaging.groovy:121]:
```suggestion
              whenTrue(isBranch()) {
```

https://github.com/elastic/apm-pipeline-library/blob/main/vars/isBranch.groovy and https://github.com/elastic/apm-pipeline-library/blob/main/vars/whenTrue.groovy for the declarative pipeline. If conditionals require some tricks with the scripts closure

**Review by v1v — APPROVED** [review]

**Pull request merged**